packages/framework/fluid-framework/src/index.ts

```diff
@@ -54,13 +54,26 @@ export type {
 	ReplaceIEventThisPlaceHolder,
 	FluidObject, // Linked in doc comment
 	FluidObjectProviderKeys, // Used by FluidObject
+	// The event APIs are known to conflict, and this is intended as the exports via `@fluidframework/core-interfaces` are preferred over the deprecated ones from `@fluidframework/tree`.
+	// eslint-disable-next-line import/export
+	Listeners,
+	// The event APIs are known to conflict, and this is intended as the exports via `@fluidframework/core-interfaces` are preferred over the deprecated ones from `@fluidframework/tree`.
+	// eslint-disable-next-line import/export
+	IsListener,
+	// The event APIs are known to conflict, and this is intended as the exports via `@fluidframework/core-interfaces` are preferred over the deprecated ones from `@fluidframework/tree`.
+	// eslint-disable-next-line import/export
+	Listenable,
+	//  The event APIs are known to conflict, and this is intended as the exports via `@fluidframework/core-interfaces` are preferred over the deprecated ones from `@fluidframework/tree`.
+	// eslint-disable-next-line import/export
+	Off,
 } from "@fluidframework/core-interfaces";
 
 export type { isFluidHandle } from "@fluidframework/runtime-utils";
 
 // Let the tree package manage its own API surface.
 // Note: this only surfaces the `@public, @beta and @alpha` API items from the tree package.
-// eslint-disable-next-line no-restricted-syntax, import/no-internal-modules
+// This reexports all non-conflicting APIs from `@fluidframework/tree`. The event APIs are known to conflict, and this is intended as the exports via `@fluidframework/core-interfaces` are preferred over the deprecated ones from `@fluidframework/tree`.
+// eslint-disable-next-line no-restricted-syntax, import/no-internal-modules, import/export
```

Comment:
You could add something like:  "When * exports conflict with named exports, the named export is used (and this lint warning is triggered). This allows the non deprecated versions of the events (above as named exports) to replace the deprecated ones from tree."

Reply:
@CraigMacomber, is there a spec / docs we can reference?
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/export says "If there are two wildcard exports statements that implicitly re-export the same name, neither one is re-exported." But I don't see the text about named exports win.

Reply:
I found that same document, and was also disappointed due to the same omission so I did some experiments.

I kinda assumed this change would have tests to make sure the exports actually work (compile and work at runtime), I guess it doesn't so those should probably be added.

Note that should the approach taken here not work, blocking the undesired exports with a second wildcard export, then adding them a third time as named exports should be fine according to that document as far as I can tell.

Reply:
I tried to understand the actual spec https://tc39.es/ecma262/multipage/ecmascript-language-scripts-and-modules.html#sec-exports but I couldn't figure it out.

Reply:
I think perhaps in https://tc39.es/ecma262/multipage/ecmascript-language-scripts-and-modules.html#sec-getexportednames step 8.d.i.1 implies it works as I claimed?

Reply:
I had also tried reading the same "sec-exports" page and could not derive the intended/prescribed behavior.
I don't see how second * wildcard would work well/easily. I think it means crafting an extra file in fluid-framework that has just the conflicts, then index exports * from that to cause specific conflicts, then also a named set.

I would favor the other version where there are no conflicts if the spec is complicated or hard to even find. I would not be surprised if some implementation got it wrong. I don't think there will be a lot of maintenance once the structure is in place (with appropriate comments for dev guidance). An alternative would be to explicitly name tree exports like we do for every other package. The other PR gives us compromise version of that option in tree itself - so engineers don't have to hop packages.

Reply:
(hmm github is not showing me all of the comments in timely fashion - reading other spec soon)

Reply:
I think yes it works as you claim, Craig, because all of 7 (IndirectExportEntries) happens before 8. What I don't see in there is support of the MDN note that two wildcards in conflict resolve to none. Per the spec, I would expect the first encountered to win. (Makes me a little concerned about just relying on spec, but not too much.) I am good with either solution with some preference for the explicit handling ourselves.
I think as tree reviewer, Craig, you will get the biggest vote.
Should we choose this and then determine there is a problem, we have an option to fix with the other pattern.